Spanner/tests/Perf/Report.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Spanner\Tests\Perf;
+
+class Report
+{
+    private $sapi;
+
+    private $message = '';
+
+    public function __construct($sapi)
+    {
+        $this->sapi = $sapi;
+    }
+
+    public static function getReporter()
+    {
+        return new self(php_sapi_name());
+    }
+
+    public function report($message)
+    {
+        if ($this->sapi === 'cli') {
+            print $message;
+        } else {
+            // Otherwise, if it is being called from a browser, aggregate into a message.
+            $this->message .= $message;
+        }
+    }
+
+    public function average(array $input)
```

Comment:
Do these methods which perform calculations need to be public? It looks like they're only used by this class.